lib/features/context-pad/ContextPad.js

```diff
@@ -259,6 +259,8 @@ ContextPad.prototype.triggerEntry = function(entryId, action, event, autoActivat
 
   var handler = entry.action;
 
+  this._eventBus.fire('contextPad.trigger', { event });
```

Comment:
Shall we allow intercepting of trigger?
```suggestion
  if (this._eventBus.fire('contextPad.trigger', { action, event }) === false) {
    return;
  };
```

Reply:
Makes sense. I added `entry` and `event` as parameters because we can get the label for the target element via `entry` (user triggered "Start Event" entry); and the type of `event` allows us infer the trigger source: drag vs click vs keyboard

Reply:
Why not expose "action", too? Given that this is public API (cf. keyboard integration) the authoritive truth should be the `action` parameter, not the event type, or am I wrong?

Reply:
We could do that too. `action` is one of "click" or "drag". Triggering via keyboard, acts as a click and therefore the `action` is click too. But by checking the event type we can actually differentiate it.

Reply:
> We could instead, send the explicit action already based on the event type. This way, listeners to it wouldn't have to work that out themselves.

Or add `keyboard` as an action, that equals `click` in behavior.

Reply:
Getting this information via the event would be something like this:
```javascript
if(event instanceOf PointerEvent) { type = "click"}
else if(event instanceOf DragEvent) {type = "drag"}
else if(event instanceOf KeyboardEvent) {type = "keyboard"}
```

Adding an `action` for keyboard means changing the providers in bpmn-js and bpmn-js-create-append-anything.
If we do want to sned `action` explicitly, it would be easier to infer that in diagram-js from the event and send it.

Reply:
Let's rely on `event` for now then